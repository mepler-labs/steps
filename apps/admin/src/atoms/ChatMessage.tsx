import React from 'react';
import styled from 'styled-components';

import Panel from './Panel';
import { white, mediumBlue } from 'styles/colors';

export enum ChatMessageType {
  sent = 'sent',
  received = 'received'
}

interface Props {
  type?: ChatMessageType;
  text: string;
  from?: string;
  color?: string;
}

class ChatMessage extends React.Component<Props, {}> {
  static props = {
    type: 'received',
    color: mediumBlue
  };

  renderChatElements() {
    const { type, text, from } = this.props;
    const chatElements = [
      <Message {...this.props} key={1}>
        {text}
      </Message>,
      <From {...this.props} key={2}>
        {from}
      </From>
    ];
    if (type === 'received') {
      return chatElements.reverse();
    }
    return chatElements;
  }
  render() {
    return <Container>{this.renderChatElements()}</Container>;
  }
}

const Container = styled.div`
  font-family: 'Tiempos', serif;
  font-size: 21px;
`;

export const Message = styled(Panel).attrs<Props>({})`
  background-color: ${({ type, color }) =>
    type === 'received' ? color : white};
  display: inline-block;
  padding: 20px;
  line-height: 25px;
`;

export const From = styled<Props, 'div'>('div')`
  display: inline-block;
  font-size: 28px;
  width: 65px;
  height: 65px;
  border-radius: 65px;
  color: ${white};
  background-color: ${({ type, color }) =>
    type === 'received' ? color : mediumBlue};
  text-align: center;
  vertical-align: middle;
  line-height: 65px;
  margin: 0 10px;
`;

export default ChatMessage;
