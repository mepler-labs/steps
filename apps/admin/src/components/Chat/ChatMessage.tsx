import React from 'react';
import styled from 'styled-components';

import Panel from 'atoms/Panel';
import { white, mediumBlue } from 'styles/colors';

type ChatMessageType = 'sent' | 'received';

interface Props {
  className?: string;
  type?: ChatMessageType;
  text: string;
  from?: string;
  color?: string;
}

class ChatMessage extends React.Component<Props, {}> {
  static props = {
    type: 'received',
    color: mediumBlue,
  };

  render() {
    const { className, type, text, from } = this.props;
    return (
      <div className={className}>
        <Panel className="message">{text}</Panel>
        <div className="from">{from}</div>
      </div>
    );
  }
}

const StyledChatMessage = styled<Props>(ChatMessage)`
    font-family: 'Tiempos', serif;
    font-size: 21px;
    display: flex;
    justify-content: flex-end;

    flex-direction: ${({ type }) =>
      type === 'received' ? 'row-reverse' : 'row'}

    .message {
      background-color: ${({ type, color }) =>
        type === 'received' ? color : white};
      display: inline-block;
      padding: 20px;
      line-height: 25px;
    }

    .from {
      display: inline-block;
      font-size: 28px;
      width: 65px;
      height: 65px;
      border-radius: 65px;
      color: ${white};
      background-color: ${({ type, color }) =>
        type === 'received' ? color : mediumBlue};
      text-align: center;
      vertical-align: middle;
      line-height: 65px;
      margin: 0 10px;
    }
  }
`;

export default StyledChatMessage;
