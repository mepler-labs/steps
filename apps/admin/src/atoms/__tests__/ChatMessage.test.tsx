import { shallow } from 'enzyme';
import ChatMessage from '../ChatMessage';
import * as React from 'react';

describe('ChatMessage.tsx', () => {
  it('renders correctly', () => {
    const wrapper = shallow(<ChatMessage text="test message" />);

    expect(wrapper).toMatchSnapshot();
  });
});
