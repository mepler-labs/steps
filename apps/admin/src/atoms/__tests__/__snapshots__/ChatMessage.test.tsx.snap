// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChatMessage.tsx renders correctly 1`] = `
<styled.div>
  <Styled(Panel)
    key="1"
    text="test message"
  >
    test message
  </Styled(Panel)>
  <styled.div
    key="2"
    text="test message"
  />
</styled.div>
`;
